### PR TITLE
Preparing many releases for new Camlp5 8.x

### PR DESCRIPTION
**Pull request by Kakadu**
continuous-integration/travis-ci/pr fails on Mac because of camlp5+perl issues there

Errors are not releveant
1) Camlp5 gives stack overflow during compilation of itself
2) Camlp5 has PERL issues on Mac

Please, review.

### CODE REVIEW

**Comment by Kakadu** [comment]
Current results:
- build passes on ubuntu+ocaml 4.11
- build fails on ubuntu + ocaml 4.09 because of crash in camlp5  
- build fails on BSD because camlp5 doesn't have right depexts about perl `IPC/System/Simple.pm` module
- same on darwin

It's OK for me if you decide to merge this PR later.

**Comment by Kakadu** [comment]
The only visible errors are from the missing perl module in Camlp5. Ready for review.

**Comment by Kakadu** [comment]
@chetmurthy I added a few comments about Perl in [this commit ](https://github.com/ocaml/opam-repository/pull/17285/commits/d421604559b6f7837f508649cf217378d5319674). I hope you are OK with this.

**Comment by chetmurthy** [comment]
Kakadu, thank you, I will add it to the upstream opam file.

**Comment by Kakadu** [comment]
> ETA: Oh, I was unable find documentation. Is it really conditionalized on the OS ?

Yeah, it is called a `global variable` in the [manual](https://opam.ocaml.org/doc/Manual.html#Global-variables)

**Comment by chetmurthy** [comment]
@Kakadu I dug around, and couldn't find anything on ... "filters for global variables" ?  Is that what they should be?  The manual is .... very confusing.  I guess I'll have to go look at source or something, to understand it.  Sigh.  The uselessness of manuals in general isn't a great incentive for those who hate writing documentation to begin with.

File: packages/GT/GT.0.3.0/opam [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+homepage: "https://github.com/JetBrains-Research/GT"
+bug-reports: "https://github.com/JetBrains-Research/GT/issues"
+
+depends: [
+  "ocaml"      { >= "4.10.0" }    
+    # 4.10 is required only because camlp5 has a few glitches in old compilers 
+    # because a few features are not properly backported (and probably will never be)
+  "camlp5"     { >= "8" | = "8.00~alpha04" | = "8.00~alpha05" }
+  "ocamlfind"  { build }
+  "dune"       { > "2.7" & build }
```
Comment (mseri) [inline packages/GT/GT.0.3.0/opam:30]:
```suggestion
  "dune"       { > "2.7" }
```

Dune should not be tagged as build dependncy for forward compatibility reasons

**Comment by mseri** [comment]
Thanks

**Pull request merged**